pyvista/utilities/reader.py

```diff
@@ -164,7 +166,6 @@ def _set_filename(self, filename):
         """Set filename and update reader."""
         # Private method since changing file type requires a
         # different subclass.
-        self.filename = filename
```

Comment:
Lets leave this change out of this PR.  You also have it implemented for the new reader class.

Reply:
> Lets leave this change out of this PR. You also have it implemented for the new reader class.

Yes, `BaseReader` sets it. Question:

1. What is meant by the comment "changing file type requires a different subclass"? I'm not sure I understand why `_set_filename()` is private.
2. Would it make more sense for me to do this instead:

```python
class DICOMReader(BaseReader):
    """DICOM Reader for reading .dcm files.

    This reader reads single files and folder (DICOM stacks).

    Examples
    --------
    >>> import pyvista as pv
    >>> from pyvista import examples
    >>> volume = examples.download_dicom_stack()
    >>> volume.plot()
    """

    _class_reader = _vtk.vtkDICOMImageReader

    def __init__(self, path: str):
        self._reader = self._class_reader()
        if os.path.isfile(path):
            self.filename = path
            self._set_filename(path)
        elif os.path.isdir(path):
            self.foldername = path
            self._set_foldername(path)

    def _set_foldername(self, foldername):
        """Set foldername and update reader."""
        self.reader.SetDirectoryName(filename)
        self._update_information()
```

Reply:
> 1. What is meant by the comment "changing file type requires a different subclass"? I'm not sure I understand why `_set_filename()` is private.

I think I was thinking about the `get_reader` usage.  I didn't want users to be confused if they tried to do something like this and expecting the reader class to change if `filename1` and `filename2` use different file types.

```py
reader = get_reader(filename1)
reader2 = reader.set_filename(filename2)
```

If there is use case to make this public, let's do it (a separate PR, lets keep them small so these PRs can finish sooner).


> 2\. Would it make more sense for me to do this instead:

Your proposal is good too.  It does require 2 method definitions instead of the 1 in your current PR.  Your proposal makes more sense to me if we are exposing these methods publicly.

Reply:
Okay, I zipped the folder and set my `PYVISTA_VTK_DATA` environment variable to a local folder containing it, and my tests pass now.